### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by Goooler**
- Remove redundant SAM.
- Proper accessors in kotlin.
- Remove unneeded generic type.
- Inline local variable.
- Const values.
- `Sort` in kotlin.
- `Until` in kotlin.

### CODE REVIEW

File: detekt-gradle-plugin/src/functionalTest/resources/jvm/src/main/kotlin/Errors.kt [hunk 1]
```diff
@@ -8,6 +8,7 @@ class Errors {
     }
 
     fun javaExit() {
+        @Suppress("ReplaceJavaStaticMethodWithKotlinAnalog")
```
Comment (cortinico) [inline detekt-gradle-plugin/src/functionalTest/resources/jvm/src/main/kotlin/Errors.kt:11]: nit: Why this? Was it triggering a warning of any form? If so where?

**Review by cortinico — APPROVED** [review]
Overall looks good 👍 Just a minor nit

File: detekt-gradle-plugin/src/functionalTest/resources/jvm/src/main/kotlin/Errors.kt (hunk 1 above)
Comment (Goooler) [reply detekt-gradle-plugin/src/functionalTest/resources/jvm/src/main/kotlin/Errors.kt:11]: As I see a suggestion by IDEA to replace `System.exit(1)` with `exitProcess(1)`, it's not proper right here?

Comment (cortinico) [reply detekt-gradle-plugin/src/functionalTest/resources/jvm/src/main/kotlin/Errors.kt:11]: > it's not proper right here?

I think it's ok. This is a test resource, so we're not actually executing it and should be treated as plain-text. However, as the IDE is recognizing that is Kotlin code, it's probably ok to have it.

**Review by schalkms — APPROVED** [review]

**Pull request merged**